### PR TITLE
feat: add request augmentation logic

### PR DESCRIPTION
**Pull request by chrismaree**
This PR is similar to the previous one submitted, see 0x6539519df26ac6a90f2a1b02b3c2f4223d6147c3e652e5f7eda663df99e9713b except it is used to find augmenting request information for L1 requests.

A sample query could look like this:

Submit the following payload to `<URL>/api/augment-request`:
```
{
  l1Requests: [
    {
      identifier: "0x5945535F4F525F4E4F5F51554552590000000000000000000000000000000000",
      time: 1662050753,
    },
  ],
};
```

and you will get back:

```
[
	{
		"l1RequestTxHash": "0xb7e990e94ecaad6d829275c19e20fcdde7f7cad7b08f94dee967ef787d81923c",
		"ooRequestUrl": "https://oracle.umaproject.org/request?transactionHash=0x6539519df26ac6a90f2a1b02b3c2f4223d6147c3e652e5f7eda663df99e9713b&chainId=137&oracleType=OptimisticOracle"
	}
]
```

This is used to find the L1 transaction hash for a given request as well as the ooRequestURL. This currently supports all the past requests for the Optimistic Oracle, Optimistic Oracle v2 and Skinny optimistic oracle on Mainnet, Polygon, boba, Optimism and arbitrum.

Note that we do NOT include ancillary data. it is not needed to correctly match between an identifier and a request and is intentionally excluded to speed up the use of this function (we dont need to send through a large payload).

### CODE REVIEW

File: pages/api/_common.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { ethers, Contract, BigNumber } from "ethers";
```
Comment (ryanwolhuter) [inline pages/api/_common.ts:1]: `BigNumber` is defined and never used, must remove to fix build failure

File: pages/api/augment-request.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  const ooChainIds = getOoChainIds();
+
+  const chainOoRequests = await Promise.all(
+    ooChainIds.map((chainId) => getOptimisticOraclePriceRequestEventsByChainId(chainId))
+  );
+  const ooRequestTxs: { requestTransactionHash: string; chainId: number; oracleType: any }[] = [];
+  l1Requests.forEach((l1Request) => {
+    let foundOoTx = { requestTransactionHash: "", chainId: 0, oracleType: "" };
+    chainOoRequests.forEach((_, ooChainId) => {
+      chainOoRequests[ooChainId].forEach((ooRequest, index) => {
```
Comment (ryanwolhuter) [inline pages/api/augment-request.ts:72]: `index` param of `forEach` is never used, must remove to fixe build error

File: pages/api/augment-request.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+      });
+    });
+    ooRequestTxs.push(foundOoTx);
+  });
+  return ooRequestTxs;
+}
+
+async function getAugmentingRequestInformation(l1Requests: Request[]) {
+  const votingV1 = await constructContractOnChain(1, "Voting");
+  //todo: add voting v2 when released.
```
Comment (ryanwolhuter) [inline pages/api/augment-request.ts:93]: lol I can't believe this, but `//todo` with no space between the `//` and `todo` causes a build error. Add a space to fix.

**Review by ryanwolhuter — APPROVED** [review]
LGTM — just a couple stupid things that caused a build error (see comments)

File: pages/api/augment-request.ts (hunk 3 above)
Comment (chrismaree) [reply pages/api/augment-request.ts:93]: thats good! this is a common linting package restriction we have in many other repos as well.

**Pull request merged**